lib/xandra/protocol.ex

```diff
@@ -18,6 +18,13 @@ defmodule Xandra.Protocol do
     %{frame | body: encode_string_map(requested_options)}
   end
 
+  def encode_request(%Frame{kind: :auth_response} = frame, requested_options, options)
+      when is_map(requested_options) do
+    [authenticator: authenticator, options: auth_options] = options[:authentication]
```

Comment:
This is something you should almost never do in Elixir. Keyword lists match the same as lists, not the same as maps, so

```elixir
options = [b: 1, a: 2]
[a: a, b: b] = options
```

fails with a `MatchError` instead of matching (like a map would in the case above). If we go with the API you implemented (we might go with the `{mod, opts}` API I commented about), you should do something along the lines of

```elixir
authenticator = Keyword.fetch!(opts[:authentication], :authenticator)
options = Keyword.fetch!(opts[:authentication], :options)

...
```